lib/components/viewers/live-stop-times.tsx

```diff
@@ -144,36 +150,88 @@ class LiveStopTimes extends Component<Props, State> {
       return stopTimeComparator(stopTimesA[0], stopTimesB[0])
     }
 
+    const routeTimes = Object.values(stopTimesByPattern)
+      .filter(({ times }) => times.length !== 0)
+      .sort(patternComparator)
+      .map((route) => {
+        return {
+          ...route,
+          times: route.times.concat()?.sort(stopTimeComparator)
+        }
+      })
+      .filter(({ pattern, route }) =>
+        routeIsValid(route, getRouteIdForPattern(pattern))
+      )
+
+    const routesByDay: any = []
+    const firstDayOfService = routeTimes[0].times[0].serviceDay
+
+    // Loop through all routes, and sort based on that route's next available day of service.
+    routeTimes.forEach((route) => {
+      const { serviceDay } = route.times[0]
+      const daysPastFirstService =
+        (serviceDay - firstDayOfService) / ONE_DAY_IN_SECONDS
+
+      // Create a separate array for each service day
+      routesByDay[daysPastFirstService]
+        ? routesByDay[daysPastFirstService].push(route)
+        : routesByDay.push([route])
+    })
```

Comment:
I'm not sure I'm crazy about `routesByDay` wrapper array. Would it not be possible to just add a `day` field to the route time object?

Reply:
Added in [e6abad9](https://github.com/opentripplanner/otp-react-redux/pull/780/commits/e6abad93592a98933c323342b67150019b12ff32)!